installer/functions

```diff
@@ -220,6 +224,41 @@ echo_tty() {
     fi
 }
 
+# Set a console color (shorthand)
+# usage: echo_color [l][color] [string]
+# Specifying "l" uses the light version of the color.
+# Color can be red, green, yellow, blue, magenta, cyan.
+# Color can be specified with just the first letter. 
+# example: echo_color "lr" "light red"
+echo_color() {
+    # If not outputting to a tty print no colors.
+    if [ ! -t 2 ]; then
+        shift
+        echo "$*" 1>&2 
+        return
+    fi
+    printf "\033["
+    local c="$1"
+    if [ "${c#l}" != "$c" ]; then
+        printf "1;"
+        c="${c#l}"
+    else
+        printf "0;"
+    fi
+    case "$c" in
+        r*) printf "31m";;
+        g*) printf "32m";;
+        y*) printf "33m";;
+        b*) printf "34m";;
+        m*) printf "35m";;
+        c*) printf "36m";;
+         *) printf "37m";;
+    esac
+    shift
+    echo "$*" 1>&2
+    printf "\033[0m"
```

Comment:
It'd be nice to squeeze this in before the newline, so that line buffering doesn't result in weird colors if multiple things are outputting to the terminal